pkg/operator/controller/externaldns/deployment.go

```diff
@@ -598,3 +602,78 @@ func buildStringMapHash(data map[string]string) (string, error) {
 	}
 	return buildMapHash(m)
 }
+
+func securityContextChanged(current, updated, desired *corev1.SecurityContext) (bool, *corev1.SecurityContext) {
```

Comment:
That was my initial implementation as well, but figured it might be better to keep it similar to `volumeMountsChanged()`.

Reply:
`volumeMountsChanged()` matches `append()` semantics as the updated slice is indeed appended inside the function, so can be reallocated. The whole container could have been passed to `volumeMountsChanged()` to append directly to it but I liked the signature with 3 slices more.

Reply:
As discussed, to update the `updated` in case it's nil the whole container has to be passed, so my proposal won't be as elegant as it was supposed to.